From 88d1b348a2ba30b9296801f7c7c959c300faac90 Mon Sep 17 00:00:00 2001
From: MSe1969 <mse1969@posteo.de>
Date: Sat, 29 Dec 2018 12:41:21 +0100
Subject: [PATCH] Different sign method in build script, remove Updater, allow
 own name

The original method, as documented by Google / LineageOS, sometimes causes
errors when executing the 'ota_from_target_files' python script.

Therefore, the combination of PRODUCT_DEFAULT_DEV_CERTIFICATE in common.mk
of vendor/lineage and setting a symlink testkey.* => releasekey.* in the
certs directory to allow selinux to build has been chosen, similar to
the docker image in github.com/lineageos4microg.

Further, the Updater apk is excluded from the build, as we don't have an
official LineageOS build, hence this app does not provide any value.

Finally, the reset option to rename all unofficial builds to UNOFFICIAL is
removed to allow a custom file name containing the expression 'signed'.

Change-Id: I4b8b26ef3f6d54f7d4ee1c570c1737779b6b8884
---
 config/common.mk | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/config/common.mk b/config/common.mk
index 23f19a16..5eda9db9 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -3,6 +3,17 @@ $(call inherit-product-if-exists, vendor/extra/product.mk)
 
 PRODUCT_BRAND ?= LineageOS
 
+# Sign with own key
+ifneq ($(OWN_KEYS_DIR),)
+PRODUCT_DEFAULT_DEV_CERTIFICATE := $(OWN_KEYS_DIR)/releasekey
+PRODUCT_OTA_PUBLIC_KEYS := $(OWN_KEYS_DIR)/releasekey
+PRODUCT_EXTRA_RECOVERY_KEYS := $(OWN_KEYS_DIR)/releasekey
+endif
+
+# Updater URI
+PRODUCT_PROPERTY_OVERRIDES += \
+    lineage.updater.uri=https://raw.githubusercontent.com/lin15-microG/OTA/lineage-15.1/$(LINEAGE_BUILD).json
+
 PRODUCT_BUILD_PROP_OVERRIDES += BUILD_UTC_DATE=0
 
 ifeq ($(PRODUCT_GMS_CLIENTID_BASE),)
@@ -264,11 +275,6 @@ ifndef LINEAGE_BUILDTYPE
     endif
 endif
 
-# Filter out random types, so it'll reset to UNOFFICIAL
-ifeq ($(filter RELEASE NIGHTLY SNAPSHOT EXPERIMENTAL,$(LINEAGE_BUILDTYPE)),)
-    LINEAGE_BUILDTYPE :=
-endif
-
 ifdef LINEAGE_BUILDTYPE
     ifneq ($(LINEAGE_BUILDTYPE), SNAPSHOT)
         ifdef LINEAGE_EXTRAVERSION
-- 
2.11.0

